File: 윤석주/day3/쇠막대기.py
def solution(arrangement):
    answer = 0
    sticks = 0
    for i in range(len(arrangement)-1):
        if arrangement[i] == '(':
            if arrangement[i+1] == ')':
                answer += sticks
            else:
                sticks += 1
                answer += 1
        else:
            if arrangement[i-1] != '(':
                sticks -= 1
    return answer

File: 박재현/이전/쇠막대기.cpp
#include <stack>
#include <string>
#include <vector>

using namespace std;

int solution(string arrangement)
{
    int answer = 0;
    stack<char> st;
    int check = 0;
    for (auto x : arrangement)
    {

        if (x == '(')
        {
            st.push('(');
            check = 0;
        }
        else
        {
            st.pop();

            if (check)
            {
                answer++;
            }
            else if (st.size() > 1)
            {
                answer += st.size();
                check = 1;
            }
        }
    }

    return answer;
}